src/cheri_insts.sail

```diff
@@ -135,6 +179,9 @@ function clause execute(CJALR(imm, cs1, cd)) = {
   if not (cs1_val.tag) then {
     handle_cheri_reg_exception(CapEx_TagViolation, cs1);
     RETIRE_FAIL
+  } else if signed(cs1_val.otype) == otype_isentry_ptp then {
+    /* Indirect points-to-PCC sentries have their own requirements; jump out */
+    handle_cjalr_isentry_ptp(cd, cs1)
   } else if isCapSealed(cs1_val) &
             ((signed(cs1_val.otype) != otype_sentry) | imm != zeros()) then {
```

Comment:
Immediate check needs to apply to indirect sentries too (as it would imply loading from a different address; currently you just silently ignore it). I do wonder though if CJALR should support these transparently or not, for Morello there are separate indirect and indirect-pair sentry branch instructions. I imagine not knowing whether or not a CJALR will need to load would be a complete pain for OoO processors as which execution unit it is issued to has a value dependency on its source register.

Reply:
Thanks for the reminder; let's follow Morello on this and we can always experiment with making `cjalr` follow transparently later.